cloudinit/net/openbsd.py

```diff
@@ -0,0 +1,41 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit import log as logging
+from cloudinit import util
+import cloudinit.net.bsd
+
+LOG = logging.getLogger(__name__)
+
+
+class Renderer(cloudinit.net.bsd.BSDRenderer):
+
+    def __init__(self, config=None):
+        super(Renderer, self).__init__()
+
+    def write_config(self):
+        for device_name, v in self.interface_configurations.items():
+            if_file = 'etc/hostname.%s' % device_name
+            fn = util.target_path(self.target, if_file)
+            with open(fn, 'w') as fd:
+                if isinstance(v, dict):
+                    fd.write(v.get('address') + ' netmask '
+                             + v.get('netmask') + '\n')
+                else:
+                    fd.write('dhcp')
+
+    def start_services(self, run=False):
+        if not self._postcmds:
+            LOG.debug("netbsd generate postcmd disabled")
+            return
+        util.subp(['sh', '/etc/netstart'], capture=True)
+
+    def set_route(self, network, netmask, gateway):
+        if network == '0.0.0.0':
+            if_file = 'etc/mygate'
+            fn = util.target_path(self.target, if_file)
```

Comment:
Can we refactor writing this file out in write_config(); that way all writes happen in a single location.